plugins/modules/azure_rm_monitoractivitylogalerts.py

```diff
@@ -0,0 +1,520 @@
+
+        self.results = dict(
+            changed=False,
+            activitylogalert=dict(),
+            diff=dict(
+                before=None,
+                after=None
+            )
+        )
+
+        super(AzureRMMonitorActivityLogAlert, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                             supports_check_mode=True,
+                                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+
+        # Defaults for variables
+        result = None
+        result_compare = dict(compare=[])
+        before_dict = None
+
+        # Get current activity log alert if it exists
+        before_dict = self.get_activity_log_alert()
+
+        # Create dict from input, without None values
+        # hhttps://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2020_10_01.models.activitylogalertresource?view=azure-python
+        activity_log_alert_template = {
+            "location": self.location,
+            "tags": self.tags,
+            "scopes": self.scopes,
+            "condition": self.condition,
+            "actions": self.actions,
+            "enabled": self.enabled,
+            "description": self.description
+        }
+        # Filter out all None values
+        activity_log_alert_input = self.filter_none(activity_log_alert_template)
+
+        # Create/Update if state==present
+        if self.state == 'present':
+            if before_dict is None:
+                # activity log alert does not exist, create
+                # On creation default to location of resource group unless otherwise noted in input variables
+                if not self.location:
+                    resource_group = self.get_resource_group(self.resource_group)
+                    activity_log_alert_input['location'] = resource_group.location
+                # On creation input == what we send to api
+                activity_log_alert_update = activity_log_alert_input
+                self.results['changed'] = True
+                if self.check_mode:
+                    # Check mode, skipping actual creation
+                    pass
+                else:
+                    create_response = self.create_or_update(activity_log_alert_update)
+            else:
+                # activity log alert already exists, updating it
+                # Dict for update is the union of existing object overwritten by input data
+                activity_log_alert_update = before_dict | activity_log_alert_input
+
+                # Enhanced with tags (special behaviour because of append_tags possibility)
+                update_tags, activity_log_alert_update['tags'] = self.update_tags(before_dict['tags'])
```

Comment:
When the value of 'tags' is None, the return value does not have 'tags'.
```suggestion
                update_tags, activity_log_alert_update['tags'] = self.update_tags(before_dict.get('tags'))
```